### PR TITLE
[LYN-9039] Object stream write response change

### PR DESCRIPTION
**Pull request by amzn-mike**
Posting @lumberyard-employee-dm fix for ObjectStreamWriteElementOverride always being expected to handle serialization.  There is now a return value enum which can direct the serializer to perform default serialization.

I've included unit tests for this new functionality.

This fixes an issue with AssetBundleManifest serialization where versions <= 2 would be written out by the override but versions newer than that were expected to be written using the default serialization but weren't, resulting in an empty manifest.

Fixes LYN-9039

### CODE REVIEW

**Review by AMZN-stankowi — APPROVED** [review]

**Review by lumberyard-employee-dm — APPROVED** [review]

File: Code/Framework/AzCore/Tests/Serialization.cpp [hunk 1]
```diff
@@ -720,6 +720,49 @@ namespace SerializeTestClasses {
...
+        static void Reflect(AZ::SerializeContext& sc)
+        {
+            sc.Class<ElementOverrideType>()
+                ->Attribute(AZ::SerializeContextAttributes::ObjectStreamWriteElementOverride, &ElementOverrideType::Writer)
+                ->Field("field", &ElementOverrideType::m_field);
+        }
+
+        int m_field = 0;
+
+        static AZ::ObjectStreamWriteOverrideResponse m_response;
```
Comment (lumberyard-employee-dm) [inline Code/Framework/AzCore/Tests/Serialization.cpp:764]: This variable isn't used.

**Comment by lumberyard-employee-dm** [comment]
Thanks for adding the UnitTest for the functionality.

**Review by lumberyard-employee-dm — APPROVED** [review]

File: Code/Framework/AzCore/Tests/Serialization.cpp [hunk 2]
```diff
@@ -1752,11 +1752,12 @@ namespace UnitTest
...
-        constexpr const char* expectedValue = R"(<ObjectStream version="3">
-	<Class name="ElementOverrideType" type="{BAA18B6C-3CB3-476C-8B41-21EA7CE1F4CF}">
-		<Class name="int" field="field" value="1" type="{72039442-EB38-4D42-A1AD-CB68F7E0EEF6}"/>
-	</Class>
-</ObjectStream>)";
+        constexpr const char* expectedValue =
+            R"(<ObjectStream version="3">)" "\n"
+            "\t" R"(<Class name="ElementOverrideType" type="{BAA18B6C-3CB3-476C-8B41-21EA7CE1F4CF}">)" "\n"
+            "\t\t" R"(<Class name="int" field="field" value="1" type="{72039442-EB38-4D42-A1AD-CB68F7E0EEF6}"/>)" "\n"
+            "\t" R"(</Class>)" "\n"
```
Comment (lumberyard-employee-dm) [inline Code/Framework/AzCore/Tests/Serialization.cpp:1759]: Why does this need tabs?

Comment (amzn-mike) [reply Code/Framework/AzCore/Tests/Serialization.cpp:1759]: The serializer outputs tabs and I'm doing a string comparison of the two so they have to match up.

Comment (lumberyard-employee-dm) [reply Code/Framework/AzCore/Tests/Serialization.cpp:1759]: StripWhiteSpace didn't get rid of it?

Comment (amzn-mike) [reply Code/Framework/AzCore/Tests/Serialization.cpp:1759]: That only removed the whitespace at the beginning/end of the string

**Pull request merged**